docs/admin/federation/index.md

```diff
@@ -0,0 +1,191 @@
+---
+---
+This guide explains how to set up cluster federation that lets us control multiple Kubernetes clusters.
+
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that we have a running Kubernetes cluster.
+If not, then head over to the [getting started guides](/docs/getting-started-guides/) to bring up a cluster.
+
+This guide also assumes that we have the Kubernetes source code that can be
+[downloaded from here](/docs/getting-started-guides/binary_release/).
+
+## Setting up a federation control plane
+
+Setting up federation requires running the federation control plane which
+consists of etcd, federation-apiserver and federation-controller-manager.
+We can run these binaries as pods on an existing Kubernetes cluster.
+
+### Getting images
+
+To run these as pods, we first need images for all the components. We can use
+official release images or we can build from HEAD.
+
+#### Using official release images
+
+As part of every release, images are pushed to `gcr.io/google_containers`. To use
+these images, we set env var `FEDERATION_PUSH_REPO_BASE=gcr.io/google_containers`
+This will always use the latest image.
+To use federation-apiserver and federation-controller-manager images from a specific release, we can set `FEDERATION_IMAGE_TAG`.
+
+#### Building and pushing images from HEAD
+
+To run the code from HEAD, we need to build and push our own images.
+We can build the images using the following command:
+
+```shell
+$ FEDERATION=true KUBE_RELEASE_RUN_TESTS=n make quick-release
+```
+
+Next, we need to push these images to a registry such as Google Container Registry or Docker Hub, so that our cluster can pull them.
+If Kubernetes cluster is running on Google Compute Engine (GCE), then we can push the images to `gcr.io/<gce-project-name>`.
+The command to push the images will look like:
+
+```shell
+$ FEDERATION=true FEDERATION_PUSH_REPO_BASE=gcr.io/<gce-project-name> ./build/push-federation-images.sh
+```
+
+### Running the federation control plane
+
+Once we have the images, we can run these as pods on our existing kubernetes cluster.
+The command to run these pods on an existing GCE cluster will look like:
+
+```shell
+$ KUBERNETES_PROVIDER=gce FEDERATION_DNS_PROVIDER=google-clouddns FEDERATION_NAME=myfederation DNS_ZONE_NAME=myfederation.example FEDERATION_PUSH_REPO_BASE=gcr.io/google_containers ./federation/cluster/federation-up.sh
```

Comment:
We need to fix `federation/manifests/federation-apiserver-deployment.yaml` before telling customers to bring up a cluster this way, since as of right now the federation etcd is not backed by persistent storage.